csvqb/csvqb/tests/unit/cube/qb/test_qb_dimension.py

```diff
@@ -0,0 +1,42 @@
+import pandas as pd
+from csvqb.models.cube.csvqb.components.dimension import NewQbDimension, NewQbCodeList
+
+
+def test_newqbdimension_extracts_newqbcodelist_newqbconcept():
+    """
+    To test a NewQbDimension correctly extracts and creates a valid NewQbCodeList & NewQbConcept
+    """
+    # New dimension with some repeated & distinct values
+    data = pd.DataFrame(
+        {
+            "New Dimension": ["A", "A", "C", "D", "E", "G", "G"]
+        }
+    )
+
+    new_dimension = NewQbDimension.from_data("Some Dataset", data["New Dimension"])
+
+    # To see if there is a code_list for the new_dimension and is of type NewQbCodeList
+    assert new_dimension.code_list is not None
+    assert isinstance(new_dimension.code_list, NewQbCodeList)
+
+    # Length of the concept list equals to number of distinct values in the data provided
+    assert len(new_dimension.code_list.concepts) == 5
+
+    # List of concept labels are asserted Individually
+    concept_labels = [c.label for c in new_dimension.code_list.concepts]
+    assert "A" in concept_labels
+    assert "C" in concept_labels
+    assert "D" in concept_labels
+    assert "E" in concept_labels
+    assert "G" in concept_labels
+
+    # assert parent_code is None
+    assert new_dimension.code_list.concepts[0].parent_code is None
```

Comment:
It would be a good idea if we could assert this for *all* concepts in the codelist.

Reply:
asserted parent is None for list of distinct concepts.